src/js/profile/charts/lineChart.js

```diff
@@ -0,0 +1,173 @@
+                value: [primaryGroup],
+            },
+            {
+                name: "tooltip",
+                value: {},
+                on: [
+                    {events: "rect:mouseover", update: "datum"},
+                    {events: "rect:mouseout", update: "{}"}
+                ]
+            },
+            {
+                name: "mainGroup",
+                value: primaryGroup,
+            },
+            {
+                name: "datatype",
+                value: {percentage: "percentage", value: "count"},
+            },
+            {
+                name: "Units",
+                value: graphValueTypes[defaultType]
+            },
+            {
+                name: "numberFormat",
+                value: {percentage: percentageFormatting, value: valueFormatting},
+            },
+            ...filterSignals
+        ],
+
+        scales: [
+            {
+                name: "xscale",
+                type: "band",
+                domain: {data: "data_formatted", field: {signal: "mainGroup"}},
+                range: "width",
+                padding: 1.5,
+                round: true
+            },
+            {
+                name: "yscale",
+                domain: {data: "data_formatted", field: {signal: "datatype[Units]"}},
+                nice: true,
+                range: "height",
+            }
+        ],
+
+        axes: [
+            {
+                orient: "bottom",
+                scale: "xscale",
+                grid: true,
+                gridOpacity: 0.5
+            },
+            {
+                orient: "left",
+                scale: "yscale",
+                labelPadding: 6,
+                format: {
+                    signal: "numberFormat[Units]"
+                }
+            }
+        ],
+
+        marks: [
+            {
+                type: "line",
+                from: {data: "data_formatted"},
+                encode: {
+                    enter: {
+                        x: {scale: "xscale", field: {signal: "mainGroup"}},
+                        width: {scale: "xscale", band: 1},
+                        y: {scale: "yscale", field: {signal: "datatype[Units]"}},
+                        y2: {scale: "yscale", value: 0}
+                    },
+                    update: {
+                        x: {scale: "xscale", field: {signal: "mainGroup"}},
+                        y: {scale: "yscale", field: {signal: "datatype[Units]"}},
+                        y2: {scale: "yscale", value: 0},
+                        tooltip: {
+                            signal: "{'percentage': format(datum.percentage, numberFormat.percentage), 'group': datum[mainGroup], 'count': format(datum.count, numberFormat.value)}"
+                        }
+                    },
+                }
+            }
+        ]
+    }
+};
+
+export const configureLinechartDownload = (data, metadata, config, annotations) => {
+    return configureLinechart(data, metadata, config);
```

Comment:
At first this surprised me. The reason configureLinechartDownload is necessary is because of how it's implemented further up in order to give some flexibility in configuring the download spec with annotations. This should be fine for future chart types

Reply:
this was missing the extra attributes when downloading a chart as an image. the code was not committed. now `configureLinechartDownload` function contains a little more than just a `return` statement but it is still much cleaner than `barChart.js`